app/uk/gov/hmrc/agentsubscriptionfrontend/models/Address.scala

```diff
@@ -29,52 +29,73 @@ case class Address(addressLine1: String,
                    postcode: Option[String],
                    countryCode: String)
 
-object Address {
+object Address{
 
   type PostCodeError = Set[String]
```

Comment:
There are some things about the validation code that strike me as a bit strange. Can we pair and try to improve them?

Some examples:
- error type is called `PostCodeError` but surely `Address.validate` may return non-postcode errors too?
- "right" type is Unit, not Address i.e. I'd expect `Validated[PostcodeError, Address]` instead of `Validated[PostCodeError, String]`